addons/account/models/account_move.py

```diff
@@ -3905,7 +3911,8 @@ def button_draft(self):
         self.write({'state': 'draft', 'is_move_sent': False})
 
     def button_cancel(self):
-        self.write({'auto_post': 'no', 'state': 'cancel'})
+        self.write({'state': 'cancel'})
+        self.write({'auto_post': 'no'})
```

Comment:
mmh that's not very elegant. Your code should be able to deal with that by doing
  `move_state = vals.get('state', move.state)`